cmake/Macros.cmake

```diff
@@ -42,11 +42,11 @@ macro(git_id RESULT)
   endif()
 endmacro(git_id)
 
-macro(use_cxx11)
+macro(use_cxx14)
   if (CMAKE_VERSION VERSION_LESS "3.1" AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    set (CMAKE_CXX_FLAGS "--std=gnu++11 ${CMAKE_CXX_FLAGS}")
+    set (CMAKE_CXX_FLAGS "--std=c++14 ${CMAKE_CXX_FLAGS}")
```

Comment:
As indicated at the [SO post](https://stackoverflow.com/questions/31965413/compile-c14-code-with-g) I quoted earlier, sounds like this won't work for gcc < v5.  All the build automated build system (and all of my own) are still on 4.x as far as I can tell.  Luckily I think we're all moved past cmake 3.1?  But then maybe we should just require 3.1 up front in the companion CMake file?

Reply:
Hm. yes not wrong, this needs to be checked (including Windows builds), if I did something wrong I will revert. But we are at c++17 now, c++14 should not be too wrong ...

Reply:
So far so good on MSVC2015 and MinGW 5.3 (gcc 5.3.0, I guess this one is >5 after all :) ).  CMake 3.10.